benchmark/opperf/nd_operations/gemm_operators.py

```diff
@@ -55,33 +57,62 @@ def run_gemm_operators_benchmarks(ctx=mx.cpu(), dtype='float32', profiler='nativ
 
     """
     # Benchmark tests for dot and batch_dot operators
-    dot_benchmark_res = run_performance_test(
-        [getattr(MX_OP_MODULE, "dot")], run_backward=True,
-        dtype=dtype, ctx=ctx,
-        inputs=[{"lhs": (1024, 1024),
-                 "rhs": (1024, 1024)},
-                {"lhs": (1000, 10),
-                 "rhs": (1000, 10),
-                 "transpose_b": True},
-                {"lhs": (1000, 1),
-                 "rhs": (100, 1000),
-                 "transpose_a": True,
-                 "transpose_b": True}],
-        warmup=warmup, runs=runs, profiler=profiler)
+    if large_tensor == "on":
+        dot_benchmark_res = run_performance_test(
+            [getattr(MX_OP_MODULE, "dot")], run_backward=True,
+            dtype=dtype, ctx=ctx,
+            inputs=[{"lhs": (2**16, 2**16),
+                     "rhs": (2**16, 2**16)},
+                    {"lhs": (4, 2**30),
+                     "rhs": (4, 2**30),
+                     "transpose_b": True},
+                    {"lhs": (2**28, 16),
+                     "rhs": (16, 2**28),
+                     "transpose_a": True,
+                     "transpose_b": True}],
+            warmup=warmup, runs=runs, profiler=profiler)
 
-    batch_dot_benchmark_res = run_performance_test(
-        [getattr(MX_OP_MODULE, "batch_dot")], run_backward=True,
-        dtype=dtype, ctx=ctx,
-        inputs=[{"lhs": (32, 1024, 1024),
-                 "rhs": (32, 1024, 1024)},
-                {"lhs": (32, 1000, 10),
-                 "rhs": (32, 1000, 10),
-                 "transpose_b": True},
-                {"lhs": (32, 1000, 1),
-                 "rhs": (32, 100, 1000),
-                 "transpose_a": True,
-                 "transpose_b": True}],
-        warmup=warmup, runs=runs, profiler=profiler)
+        batch_dot_benchmark_res = run_performance_test(
+            [getattr(MX_OP_MODULE, "batch_dot")], run_backward=True,
+            dtype=dtype, ctx=ctx,
+            inputs=[{"lhs": (1, 2**16, 2**16),
+                     "rhs": (1, 2**16, 2**16)},
+                    {"lhs": (1, 4, 2**30),
+                     "rhs": (1, 4, 2**30),
+                     "transpose_b": True},
+                    {"lhs": (1, 2**28, 16),
+                     "rhs": (1, 16, 2**28),
+                     "transpose_a": True,
+                     "transpose_b": True}],
+            warmup=warmup, runs=runs, profiler=profiler)
+    else:
```

Comment:
It seems the only difference between the if and else branch is the `inputs` argument. Can we only generate different inputs in the if/else branch and pass them to the same operator function?